code/go/internal/validator/folder_item_spec.go

```diff
@@ -78,6 +78,7 @@ func (s *folderItemSpec) validate(fs http.FileSystem, folderSpecPath string, ite
 
 	// validation with schema
 	documentLoader := gojsonschema.NewBytesLoader(itemData)
+	refreshFormatCheckersContext(filepath.Dir(itemPath))
```

Comment:
Given that we are essentially creating a new custom format checker for every `filepath.Dir(itemPath)`, perhaps we should also remove this format checker using `defer gojsonschema.FormatCheckers.Remove(...)`?

Additionally I would add a comment here noting that this call is not thread-safe so, in the future, if we need to introduce concurrency within a single `elastic-package lint` invocation, we can add mutexes here or something like that.

Reply:
I can add a mutex to surround this place, but it's just mitigation.

Just wanted to make sure that we understand the risk and consequences here (incl. error messages).

Reply:
Up to you if you want to add the mutex now or later. I don't think it's going to make much of a performance difference to have it now itself but if you want to KISS feel free to leave it out until we need it.

Reply:
Ok, it wasn't hard to add the mutex, so I did it. I've also adjusted the error message.